Description: Temporarily downgrade dep on blake3
 Newer just depends on blake3 1.5 +mmap+rayon, particularly using
 blake3::Hasher::update_mmap_rayon(), which is absent in packaged blake3 1.3.
 OTOH, updating blake3 to 1.5 involves constant_time_eq 0.1 -> 0.3 and a bunch
 of other rdeps. After these are resolved, this can be removed.
Forwarded: not-needed
Last-Update: 2024-06-12
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -64,7 +64,5 @@
 [dependencies.blake3]
-version = "1.5.0"
-features = [
-    "rayon",
-    "mmap",
-]
+version = "1.3.0"
+[dependencies.memmap2]
+version = "0.9"
 
--- a/src/function.rs
+++ b/src/function.rs
@@ -173,5 +173,8 @@
   let mut hasher = blake3::Hasher::new();
-  hasher
-    .update_mmap_rayon(&path)
+  let f = std::fs::File::open(&path)
+    .map_err(|err| format!("Failed to hash `{}`: {err}", path.display()))?;
+  let mmap = unsafe { memmap2::Mmap::map(&f) }
     .map_err(|err| format!("Failed to hash `{}`: {err}", path.display()))?;
+  hasher
+    .update(&mmap);
   Ok(hasher.finalize().to_string())
